commit aeb7ea57c9a548dd1489cf7bd6566a16ec396415 (refs/changes/07/1907/2)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-05-04T16:07:51-07:00 (2 years, 5 months ago)
    
    HEAD-2360 build.spec.local "zones" build artefacts could support a *channel* when pulling from updates.jo
    Reviewed by: Joshua M. Clulow <jmc@joyent.com>
    Approved by: Joshua M. Clulow <jmc@joyent.com>

diff --git a/README.md b/README.md
index 8a4bcb05..edf257ba 100644
--- a/README.md
+++ b/README.md
@@ -212,7 +212,8 @@ UUID, e.g.
     "zones": {
         "adminui": {
             "source": "imgapi",
-            "uuid": "ef967904-fd86-11e4-9c90-2bbf99b9e6cf"
+            "uuid": "ef967904-fd86-11e4-9c90-2bbf99b9e6cf",
+            "channel": "experimental"   // optional IMGAPI channel
         },
         ...
     },
diff --git a/buildtools/novus/cmd/downloader.js b/buildtools/novus/cmd/downloader.js
index 2c50d117..9ea7e6c8 100644
--- a/buildtools/novus/cmd/downloader.js
+++ b/buildtools/novus/cmd/downloader.js
@@ -211,6 +211,7 @@ bit_enum_zone(be, next)
 			bfi_prefix: 'zone.' + name,
 			bfi_imgapi: 'https://updates.joyent.com',
 			bfi_uuid: zone_spec('uuid'),
+			bfi_channel: zone_spec('channel'),
 			bfi_name: jobname
 		}, next);
 		return;
diff --git a/buildtools/novus/lib/bits_from/image.js b/buildtools/novus/lib/bits_from/image.js
index ba0d0194..3abf5274 100755
--- a/buildtools/novus/lib/bits_from/image.js
+++ b/buildtools/novus/lib/bits_from/image.js
@@ -31,6 +31,7 @@ bits_from_image(out, bfi, next)
 	mod_assert.string(bfi.bfi_uuid, 'bfi.bfi_uuid');
 	mod_assert.string(bfi.bfi_name, 'bfi.bfi_name');
 	mod_assert.optionalString(bfi.bfi_version, 'bfi.bfi_version');
+	mod_assert.optionalString(bfi.bfi_channel, 'bfi.bfi_channel');
 	mod_assert.func(next, 'next');
 
 	var uuid = bfi.bfi_uuid;
@@ -41,8 +42,9 @@ bits_from_image(out, bfi, next)
 		'images',
 		bfi.bfi_uuid
 	].join('/');
+	var query = (bfi.bfi_channel ? '?channel=' + bfi.bfi_channel : '');
 
-	lib_common.get_json_via_http(url, function (err, img) {
+	lib_common.get_json_via_http(url + query, function (err, img) {
 		if (err) {
 			next(new VError(err, 'could not get image "%s"',
 			    uuid));
@@ -112,7 +114,7 @@ bits_from_image(out, bfi, next)
 				'.zfs.',
 				fil.compression === 'bzip2' ? 'bz2' : 'gz'
 			].join('')),
-			bit_url: url + '/file',
+			bit_url: url + '/file' + query,
 			bit_hash_type: 'sha1',
 			bit_hash: fil.sha1,
 			bit_size: fil.size,
